src_c/_sdl2/video.pyx

```diff
@@ -940,3 +966,57 @@ cdef class Renderer:
         if not dest:
             return self.get_viewport()
         return dest
+
+    def draw_line(self, p1, p2):
+        # https://wiki.libsdl.org/SDL_RenderDrawLine
+        res = SDL_RenderDrawLine(self._renderer,
+                                 p1[0], p1[1],
+                                 p2[0], p2[1])
+        if res < 0:
+            raise error()
+
+    def draw_point(self, point):
+        # https://wiki.libsdl.org/SDL_RenderDrawPoint
+        res = SDL_RenderDrawPoint(self._renderer,
+                                  point[0], point[1])
+        if res < 0:
+            raise error()
+
+    def draw_rect(self, rect):
+        # https://wiki.libsdl.org/SDL_RenderDrawRect
+        cdef SDL_Rect _rect = pgRect_AsRect(rect)
+        res = SDL_RenderDrawRect(self._renderer, &_rect)
+        if res < 0:
+            raise error()
+
+    def fill_rect(self, rect):
+        # https://wiki.libsdl.org/SDL_RenderFillRect
+        cdef SDL_Rect _rect = pgRect_AsRect(rect)
+        res = SDL_RenderFillRect(self._renderer, &_rect)
+
+        if res < 0:
+            raise error()
+
+    def get_surface(self, depth=32):
+        # https://wiki.libsdl.org/SDL_RenderReadPixels
+        """
+            Read pixels from the current rendering target and create a pygame.Surface.
+            WARNING: This is a very slow operation, and should not be used frequently.
+        """
+        cdef Uint32 format
+        try:
+            format = format_from_depth(depth)
+        except ValueError as e:
+            raise e
+
+        cdef SDL_Texture* tex = SDL_GetRenderTarget(self._renderer)
+        cdef int width, height
+        if tex == NULL:
+            # NULL for default target (the window that created the renderer)
+            width, height = self._win.size
+        else:
+            SDL_QueryTexture(tex, &format, NULL, &width, &height)
```

Comment:
SDL_QueryTexture can fail

Reply:
This will be done with surface size & format check.